pkg/cli/config/kubeletbootstrapkubeconfig/new_kubelet_bootstrap_kubeconfig.go

```diff
@@ -0,0 +1,158 @@
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete())
+			cmdutil.CheckErr(o.Validate(args))
+			r, err := o.ToRuntime()
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(r.Run(context.TODO()))
+
+		},
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *NewKubeletBootstrapKubeconfigOptions) AddFlags(cmd *cobra.Command) {
+}
+
+func (o *NewKubeletBootstrapKubeconfigOptions) Complete() error {
+
+	return nil
+}
+
+func (o NewKubeletBootstrapKubeconfigOptions) Validate(args []string) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no arguments allowed")
+	}
+
+	return nil
+}
+
+func (o *NewKubeletBootstrapKubeconfigOptions) ToRuntime() (*NewKubeletBootstrapKubeconfigRuntime, error) {
+	clientConfig, err := o.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+	configClient, err := configv1client.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &NewKubeletBootstrapKubeconfigRuntime{
+		KubeClient:   kubeClient,
+		ConfigClient: configClient,
+		IOStreams:    o.IOStreams,
+	}
+
+	return ret, nil
+}
+
+type NewKubeletBootstrapKubeconfigRuntime struct {
+	KubeClient   kubernetes.Interface
+	ConfigClient configv1client.Interface
+
+	genericclioptions.IOStreams
+}
+
+func (r *NewKubeletBootstrapKubeconfigRuntime) Run(ctx context.Context) error {
+	infrastructure, err := r.ConfigClient.ConfigV1().Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to read infrastructure: %w", err)
+	}
+
+	bootstrapTokenSecret, err := r.KubeClient.CoreV1().Secrets("openshift-machine-config-operator").Get(ctx, "node-bootstrapper-token", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to read bootstrap token: %w", err)
+	}
+	token := bootstrapTokenSecret.Data["token"]
+	if len(token) == 0 {
+		return fmt.Errorf("token is missing from the secret: %w", err)
+	}
+
+	serverCABundle, err := refreshcabundle.GetCABundleToTrustKubeAPIServer(ctx, r.KubeClient)
+	if err != nil {
+		return fmt.Errorf("unable to get the CA bundle from the cluster: %w", err)
+	}
+
+	newConfig := clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{
+			"cluster": {
+				Server:                   infrastructure.Status.APIServerInternalURL,
+				CertificateAuthorityData: []byte(serverCABundle),
+			},
+		},
+		AuthInfos: map[string]*clientcmdapi.AuthInfo{
+			"bootstrap": {
+				Token: string(token),
+			},
+		},
+		Contexts: map[string]*clientcmdapi.Context{
+			"bootstrap": {
+				Cluster:  "cluster",
+				AuthInfo: "bootstrap",
+			},
+		},
+		CurrentContext: "bootstrap",
+	}
+	newKubeletBootstrapConfig, err := clientcmd.Write(newConfig)
+	if err != nil {
+		return fmt.Errorf("unable to serialize new kubeconfig: %w", err)
+	}
+
+	fmt.Fprintln(r.Out, string(newKubeletBootstrapConfig))
```

Comment:
Under what scenario does the admin need to see this directly instead of having us directly rotate the file on the node?

I'm OK creating two separate primitives here to be clear, I just don't quite understand why we are instead of having a more opinionated wired-together flow.

Reply:
> Under what scenario does the admin need to see this directly instead of having us directly rotate the file on the node?
>
> I'm OK creating two separate primitives here to be clear, I just don't quite understand why we are instead of having a more opinionated wired-together flow.

if our rotation command fails, having this available allows us to do the hard part (content of the file), while allowing any distribution method.  Perhaps our "put it on the nodes" fails and we need to re-run it.  Or perhaps its doomed in some scenario and they need to SSH it. This gives us those options.